receiver/prometheusexecreceiver/receiver_test.go

```diff
@@ -71,56 +69,53 @@ func execErrorTest(t *testing.T, errorReceiverConfig configmodels.Receiver) {
 func endToEndScrapeTest(t *testing.T, receiverConfig configmodels.Receiver, testName string) {
 	sink := &exportertest.SinkMetricsExporter{}
 	wrapper, err := new(component.ReceiverCreateParams{Logger: zap.NewNop()}, receiverConfig.(*Config), sink)
-	if err != nil {
-		t.Errorf("end_to_end_test.go got error = %w", err)
-	}
+	assert.NoError(t, err, "new() returned an error")
 
 	ctx := context.Background()
 	err = wrapper.Start(ctx, componenttest.NewNopHost())
-	if err != nil {
-		t.Errorf("end_to_end_test.go got error = %w", err)
-	}
-	defer wrapper.Shutdown(ctx)
+	assert.NoError(t, err, "Start() returned an error")
+	defer func() { assert.NoError(t, wrapper.Shutdown(ctx)) }()
 
 	var metrics []pdata.Metrics
 
 	// Make sure a first scrape works by checking for metrics in the test metrics exporter "sink", only return true when there are metrics
 	const waitFor = 15 * time.Second
-	const tick = 500 * time.Millisecond
+	const tick = 100 * time.Millisecond
 	require.Eventuallyf(t, func() bool {
 		got := sink.AllMetrics()
-		return len(got) != 0
+		if len(got) != 0 {
+			metrics = got
+			return true
+		}
+		return false
```

Comment:
For consistency, it'd be nice to make the order of `if X return true/false` the same here as in line 96

i.e.
```suggestion
		if len(got) == 0 {
			return false
		}

		metrics = got
		return true
```

Reply:
On second thoughts, do we even need two `Eventually` calls here? Can't we just do one and wait until `len(sink.AllMetrics()) >= 2` (which must imply the application has crashed and restarted iiuc?